gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -728,6 +729,39 @@ protected void postProcessTaskStates(@SuppressWarnings("unused") List<TaskState>
    */
   protected void postProcessJobState(JobState jobState) {
     postProcessTaskStates(jobState.getTaskStates());
+    if (!GobblinMetrics.isEnabled(this.jobProps)) {
+      return;
+    }
+    List<DatasetTaskSummary> datasetTaskSummaries = new ArrayList<>();
+    Map<String, JobState.DatasetState> datasetStates = this.jobContext.getDatasetStatesByUrns();
+    // Only process successful datasets unless configuration to process failed datasets is set
+    for (JobState.DatasetState datasetState : datasetStates.values()) {
+      if (datasetState.getState() == JobState.RunningState.COMMITTED
+        || (datasetState.getState() == JobState.RunningState.FAILED && this.jobContext.getJobCommitPolicy() == JobCommitPolicy.COMMIT_SUCCESSFUL_TASKS)) {
+        long totalBytesWritten = 0;
+        long totalRecordsWritten = 0;
+        for (TaskState taskState : datasetState.getTaskStates()) {
+          if ((taskState.getWorkingState() == WorkUnitState.WorkingState.COMMITTED
+                || PropertiesUtils.getPropAsBoolean(jobProps, ConfigurationKeys.WRITER_COUNT_METRICS_FROM_FAILED_TASKS, "false"))
+              && taskState.contains(ConfigurationKeys.WRITER_BYTES_WRITTEN)
+              && taskState.contains(ConfigurationKeys.WRITER_RECORDS_WRITTEN)) {
+            totalBytesWritten += taskState.getPropAsLong(ConfigurationKeys.WRITER_BYTES_WRITTEN);
+            totalRecordsWritten += taskState.getPropAsLong(ConfigurationKeys.WRITER_RECORDS_WRITTEN);
+          }
+        }
+        LOG.info("Reporting that " + totalRecordsWritten + " records and " + totalBytesWritten + " bytes were written for " + datasetState.getDatasetUrn());
```

Comment:
I suggest a more parseable, more "grep-able" and less conversational format in case we ever want to audit events against logging.
e.g.
```
"DatasetMetrics for '%s' - (records: %d; bytes: %d)"
```